hooks/get_badge.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) 2019 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+
+import sgtk
+from sgtk.platform.qt import QtGui
+
+HookClass = sgtk.get_hook_baseclass()
+
+
+class GetBadge(HookClass):
+    """
+    Hook that can be used to generate badges for display on publishes and work files.
+    """
+
+    def get_publish_badge(self, publish_details, publish_path, **kwargs):
+        """
+        Generate a badge for a publish.
+
+        :param publish_details:         Dictionary
```

Comment:
the right syntax to employ here is `:param dict publish_details` and `:param str publish_path`

The types will be properly referenced in the Sphinx doc.

Reply:
✅